src/components/Wizard/CreateVmWizard/providers/index.js

```diff
@@ -29,12 +43,17 @@ export const importProviders = (basicSettings, operatingSystems, WithResources,
     choices: providerList,
     required: true,
     onChange: (...opts) => onProviderChanged(k8sCreate, k8sGet, ...opts),
-    isVisible: basicVmSettings => isImageSourceType(basicVmSettings, PROVISION_SOURCE_IMPORT),
+    isVisible: basicVmSettings =>
+      settingsValue(basicSettings, NAMESPACE_KEY) && isImageSourceType(basicVmSettings, PROVISION_SOURCE_IMPORT),
```

Comment:
Likewise - can become unmaintainable if not resolved at single place

Reply:
yes, I am aware of that. I will do a large refactoring of BasicSettingsTab and vmWare provider in the followup. It is becoming quite unmaintainable